src/cdk/tree/tree.ts

```diff
@@ -205,12 +300,12 @@ export class CdkTree<T, K = T>
    * relative to the function to know if a node should be added/removed/moved.
    * Accepts a function that takes two parameters, `index` and `item`.
    */
-  @Input() trackBy: TrackByFunction<T>;
+  readonly trackBy = input<TrackByFunction<T>>();
 
   /**
    * Given a data node, determines the key by which we determine whether or not this node is expanded.
    */
-  @Input() expansionKey?: (dataNode: T) => K;
+  readonly expansionKey = input<(dataNode: T) => K>();
```

Comment:
Changing these to signals is a breaking change since signal inputs aren't writable. If signals make the rest of the logic easier, we can create a private signals with a setter/getter on the input.

Reply:
fixed